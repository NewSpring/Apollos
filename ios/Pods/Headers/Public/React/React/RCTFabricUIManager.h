../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h