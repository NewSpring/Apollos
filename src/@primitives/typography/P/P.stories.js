import React from 'react';

import { storiesOf } from '@storybook/react-native';

import ThemeProvider from '@primitives/ThemeProvider';
import P from './';

storiesOf('@primitives/typography', module)
  .add('P', () => (
    <ThemeProvider>
      <P>{'"True faith means holding nothing back. It means putting every hope in God\'s fidelity to His Promises." â€• Francis Chan'}</P>
    </ThemeProvider>
  ));
