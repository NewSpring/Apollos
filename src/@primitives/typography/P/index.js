import React from 'react';
import { Text } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import PropTypes from 'prop-types';
import withTheme from '@primitives/withTheme';

const enhance = compose(
  pure,
  withTheme(({ theme: { defaultTextColor } }) => ({ defaultTextColor })),
  setPropTypes({
    children: PropTypes.node,
    style: Text.propTypes.style,
    // fontFamily: PropTypes.oneOf(['serif', 'sansSarif']),
  }),
);

const P = enhance(({
  children,
  style: styleProp = {},
  // fontFamily = 'serif',
  defaultTextColor,
  ...otherProps
}) => {
  const tempStyle = { // TODO: replace this with passing theme props into our styled HOC
    // fontFamily,
    color: defaultTextColor,
  };

  return (
    <Text
      style={[tempStyle, styleProp]}
      {...otherProps}
    >
      {children}
    </Text>
  );
});

export default P;
