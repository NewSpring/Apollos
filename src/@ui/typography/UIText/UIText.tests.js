import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';
import P from './';

describe('the P component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Providers>
        <P>Default P text</P>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <P style={salmon}>Salmon text</P>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <Providers>
        <P accessible={false}>{'"True faith means holding nothing back. It means putting every hope in God\'s fidelity to His Promises." â€• Francis Chan'}</P>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
