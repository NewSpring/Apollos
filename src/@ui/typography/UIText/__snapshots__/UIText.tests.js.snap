// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the P component should accept additional props 1`] = `
<Text
  accessible={false}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#303030",
      "fontFamily": null,
      "fontSize": 18,
      "lineHeight": 20,
    }
  }
>
  "True faith means holding nothing back. It means putting every hope in God's fidelity to His Promises." â€• Francis Chan
</Text>
`;

exports[`the P component should accept and render passed in styles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "salmon",
      "fontFamily": null,
      "fontSize": 18,
      "lineHeight": 20,
    }
  }
>
  Salmon text
</Text>
`;

exports[`the P component should render correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#303030",
      "fontFamily": null,
      "fontSize": 18,
      "lineHeight": 20,
    }
  }
>
  Default P text
</Text>
`;
